doc/modules/computing.rst

```diff
@@ -504,20 +504,138 @@ Links
+that it's using.
+
+Scikit-learn generally relies on the ``loky`` backend, which is joblib's
+default backend. Loky is a multi-processing backend. To avoid duplicating
+the memory in each process (which isn't reasonable with big datasets),
+``loky`` will create a `memmap
+<https://docs.scipy.org/doc/numpy/reference/generated/numpy.memmap.html>`_
+that all processes can share, when the data is bigger than 1MB.
+
+In some specific cases (when the code that is run in parallel releases the
+GIL), scikit-learn will indicate to ``joblib`` that a multi-threading
+backend is preferable.
+
+As a user, you may control the backend that joblib will use (regardless of
+what scikit-learn recommends) by using a context manager::
+
+    from joblib import parallel_backend
+
+    with parallel_backend('threading', n_jobs=2):
+        # Your scikit-learn code here
+
+Please refer to the `joblib's docs <https://joblib.readthedocs.io/en/latest/>`_
+for more details.
+
+In practice, whether parallelism is helpful at improving runtime depends on
+many factors. It is usually a good idea to experiment rather than assuming
+that increasing the number of jobs is always a good thing. In some cases it
+can be highly detrimental to performance to run multiple copies of some
+estimators or functions in parallel (see oversubscription below).
+
+The nested use of joblib-based parallelism with the same backend will result
+in an exception. For example, ``GridSearchCV(OneVsRestClassifier(SVC(),
+n_jobs=2), n_jobs=2)`` won't work.
+
+OpenMP-based parallelism
+........................
+
+OpenMP is used to parallelize code written in Cython or C, relying on
+multi-threading exclusively. By default (and unless joblib is trying to
+avoid oversubscription), the implementation will use as many threads as
+possible.
+
+You can control the exact number of threads that are used via the
+``OMP_NUM_THREADS`` environment variable::
+
+    OMP_NUM_THREADS=4 python my_script.py
+
+Parallel Numpy routines from numerical libraries
+................................................
+
+Scikit-learn relies heavily on Numpy, which may be configured to use a
+multi-threaded numerical processor library like MKL or OpenBLAS.
+
+The number of threads used by the BLAS or MKL libraries can be set via the
+``MKL_NUM_THREADS`` and ``OPENBLAS_NUM_THREADS`` environment variables.
+
+Please note that scikit-learn has no direct control over these
+implementations. Scikit-learn solely relies on Numpy.
+
+Oversubscription: spawning too many threads
+...........................................
+
+It is generally recommended to avoid using significantly more processes or
+threads than the number of CPUs on a machine. Over-subscription happens when
+you're trying to spawn too many threads at the same time.
+
+Suppose you have a machine with 4 CPUs. Consider a case where you're running
+a :class:`~GridSearchCV` (parallelized with joblib) with ``n_jobs=8`` over
```

Comment:
in practice nobody will set `n_jobs=8` on a 4 cpus machine. Maybe replace it with a 8 cpus machine ?